src/shared/ui/GenreSelect/GenereSelect.test.js

```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import '@testing-library/jest-dom';
+import { render, screen } from "@testing-library/react";
+import userEvent from '@testing-library/user-event'
+import { GenreSelect } from "./GenreSelect";
+import { genreList } from "./GenreMockData";
+
+describe("GenreSelect Component Functionality", () => {
+    test("Should render all genres", () => {
+        const OnGenreClick = jest.fn();
+        render(
+            <GenreSelect
+                genres={genreList}
+                onSearch={OnGenreClick}
+            />
+        );
+        genreList.map((genre) =>
+            expect(screen.getByText(genre)).toBeInTheDocument()
+        );
+    });
+
+    test("Should select genre on button click event", () => {
+        const OnGenreClick = jest.fn((x) => x);
```

Comment:
we don't need that for mocked function
```suggestion
        const OnGenreClick = jest.fn();
```

Reply:
removed